packages/build-utils/src/types.ts

```diff
@@ -491,3 +495,28 @@ export type ShouldServe = (
 export type StartDevServer = (
   options: StartDevServerOptions
 ) => Promise<StartDevServerResult>;
+
+/**
+ * TODO: The following types will eventually be exported by a more
+ *       relevant package.
+ */
+type VariantJSONArray = ReadonlyArray<VariantJSONValue>;
+
+type VariantJSONValue =
+  | string
+  | boolean
+  | number
+  | null
+  | VariantJSONArray
+  | { [key: string]: VariantJSONValue };
+
+type VariantOption = {
+  value: VariantJSONValue;
+  label?: string;
+};
+
+export interface VariantDefinition {
+  options?: VariantOption[];
+  url?: string;
```

Comment:
@AndyBitz let's rename this property to "origin" so we can in the future extend it to something like

`origin: string | { name: string, url: string }`, which would not make sense if it's called "url"

Reply:
@dferber90 I'll do that in a follow up PR so that we can get the initial changes out sooner.

The types must come from a different package eventually anyways, so they shouldn't be blocking here.